src/main/java/net/minecraftforge/common/extensions/IForgeItemStack.java

```diff
@@ -384,6 +396,15 @@ default CompoundNBT getShareTag()
      */
     default void readShareTag(@Nullable CompoundNBT nbt)
     {
+        if (nbt != null && nbt.contains("ForgeCaps", Constants.NBT.TAG_COMPOUND))
+        {
+            this.deserializeShareTag(nbt.getCompound("ForgeCaps"));
+            nbt.remove("ForgeCaps");
+            if (nbt.size() == 0)
+            {
+                return;
```

Comment:
Why does this early exist if no forge caps are present?

Reply:
It exists early if the NBT compound is empty after reading the forge caps.